cupyx/scipy/sparse/_data.py

```diff
@@ -227,7 +225,7 @@ def _arg_min_or_max(self, axis, out, op, compare):
             if self.nnz == 0:
                 return 0
             else:
-                zero = self.dtype.type(0)
+                zero = cupy.asarray(self.dtype.type(0))
```

Comment:
Is this enough to prevent synchronization?

Reply:
No, the result of `compare(m, zero)` is a `cupy.ndarray`, and it will be converted to a CPU bool for use as a condition of if-statement. So, we need to branch on the GPU process by using `cupy.which`, just as we implement in `_find_missing_index`.

Reply:
@asi1024, it seems that this is the only place where a zero of a certain dtype is being defined, my guess here is that the test inputs are `int64`, which produces `int64` outputs as well. Do you want to restrict the output to `int32`?

Reply:
Do you mean this failure is due to a bug in SciPy? If so, I think it is OK to skip type check by feeding `type_check=False` to the testing decorator.
https://docs.cupy.dev/en/stable/reference/generated/cupy.testing.numpy_cupy_array_equal.html#cupy.testing.numpy_cupy_array_equal

Reply:
In Windows, calls to `argmax` on a SciPy sparse matrix return a scalar number, as opposed to an `np.array`:

```
>>> a = np.random.randint(2, size=(10, 20))
>>> x = sp.csc_matrix(a)
>>> x.dtype
dtype('int32')
>>> x.argmax(axis=None)
0
>>> x.astype(np.int64).argmax(axis=None)
0
```

Reply:
So, we can add `type_check=False`